mp/src/gameui/OptionsSubMouse.cpp

```diff
@@ -46,37 +46,19 @@ COptionsSubMouse::COptionsSubMouse(vgui::Panel *parent) : PropertyPage(parent, N
         "m_rawinput");
 
 	m_pMouseSensitivitySlider = new CvarSlider( this, "Slider", "#GameUI_MouseSensitivity",
-        0.0001f, 20.0f, "sensitivity", true);
+        0.0001f, 20.0f, "sensitivity", true, true);
 
-    m_pMouseSensitivityLabel = new TextEntry(this, "SensitivityLabel");
-    m_pMouseSensitivityLabel->AddActionSignalTarget(this);
+    m_pMouseSensitivityEntry = new CvarTextEntry(this, "SensitivityLabel", "sensitivity", "%.3f");
+    m_pMouseSensitivityEntry->AddActionSignalTarget(this);
+    m_pMouseSensitivityEntry->SetAllowNumericInputOnly(true);
 
-    m_pMouseAccelLabel = new TextEntry(this, "MouseAccelerationLabel");
-    m_pMouseAccelLabel->AddActionSignalTarget(this);
+    m_pMouseAccelEntry = new CvarTextEntry(this, "MouseAccelerationLabel", "m_customaccel_exponent", "%.3f");
+    m_pMouseAccelEntry->AddActionSignalTarget(this);
+    m_pMouseAccelEntry->SetAllowNumericInputOnly(true);
     m_pMouseAccelToggle = new CheckButton(this, "MouseAccelerationCheckbox", "#GameUI_MouseAcceleration");
-    m_pMouseAccelSlider = new CvarSlider(this, "MouseAccelerationSlider", "#GameUI_MouseAcceleration", 1.0f, 20.0f, "m_customaccel_exponent", true);
+    m_pMouseAccelSlider = new CvarSlider(this, "MouseAccelerationSlider", "#GameUI_MouseAcceleration", 1.0f, 20.0f, "m_customaccel_exponent", true, true);
```

Comment:
I feel as though CvarSlider should use the given cvar's min and max and not have to require it be passed in... though this is something to tackle in another branch.

Reply:
Something similar to the `CvarComboBox` would be good too. It still accepts the the number of items even though it uses only the cvar min/max.

Reply:
Yeah absolutely.

Reply:
And not only that, but also the slider shouldn't need an extra constructor param to automatically apply.